### ISSUE TITLE
Bugs in stat box

### ISSUE THREAD

**Issue by Moelf** [Momox357/Atas-OpenData-Notebooks-Julia#1]
we only added `nentries` recently after realizing we don't have this functionality, the approach used in your notebook would be incorrect as soon as the entries are weighted
- https://github.com/Moelf/FHist.jl/commit/05ab85b492b2cf2a08aeee64c1677a5115168fbf

as for plotting, we've been adding Makie recipe through `Requires.jl`, here's an minimal example:
```julia
using CairoMakie, FHist

h1 = Hist1D(randn(1000), FHist.StatsBase.Weights(rand(1000)), -3:0.3:3)

p = stairs(h1)
FHist.statbox!(p, h1)
```
![image](https://user-images.githubusercontent.com/5306213/173088904-ee8b62d9-954b-4e2b-bf7d-5099f6ec23f4.png)

or 
```julia
p = plot(h1)
FHist.statbox!(p, h1)
```
![image](https://user-images.githubusercontent.com/5306213/173088960-fbfc27cb-8a3e-4c4b-ad7a-7d3dd5ce584a.png)

while we're at it, here's an example to stacked histograms with error bars:
```julia
h1 = Hist1D(randn(1000), -3:0.3:3)
h2 = Hist1D(randn(10000), -3:0.3:3)

fig, a, p = stackedhist([h1, h2, h2 ]; errors=true) #errors defaults to `true`

labels = ["ZZ", "Z+jets", "ttbarZ"]
elements = [PolyElement(polycolor = p.attributes.color[][i]) for i in 1:length(labels)]
title = "Process names"

Legend(fig[1,2], elements, labels, title)
fig
```
![image](https://user-images.githubusercontent.com/5306213/173089102-378badb1-c229-46b2-90d9-41656990315b.png)

**Comment by MTerwissen** [comment]
Thank you for the really helpful feedback!
My approach to adding the the statistics it was more of a "workarround" way since the only alternative I found was doing it via anotations which looked even worse.

I'll definitely will have a look at Makie now!
Many thanks for the example with the stacked plots as well since I've used the [recipe](https://github.com/JuliaPlots/StatsPlots.jl/pull/315) you had submitted to StatsBase.jl so far.
The implementation for stacked (bar) plots in Plots.jl (as mentioned in the documentation with a warning that it may not be fully implemented) didn't seem to work  and StatsPlots.jl's way of doing things with @df isn't meant to work with Statsbase / FHist histograms as far as I understand it.

**Comment by Moelf** [comment]
Yeah, Plots.jl is too messy I couldn't get a clean composable interface to work.

Makie is seems to be the future so to speak, and more flexible that can probably produce publication quality plot that can be ROOT plot imposter even ;)

Feel free to let me know if you have any plot style you want to reproduce from ATLAS open analysis

**Comment by Moelf** [comment]
here's a few more that works on #master branch of FHist.jl
![image](https://user-images.githubusercontent.com/5306213/173164578-f43555b8-3ef9-4dbc-a92e-3dee8f4e2ff5.png)
![image](https://user-images.githubusercontent.com/5306213/173164596-1afff1ca-3f6c-43e4-bffe-675be9197a0a.png)

**Comment by Moelf** [comment]
this issue is becoming my log book:
```julia
h1 = Hist1D(randn(10000), -3:0.1:3)
h2 = Hist1D(randn(10000) .* (1 .+ rand.()), -3:0.1:3)
with_theme(ATLASTHEME) do
    f, a, p = stairs(h1; label="h1")
    stairs!(a, h2; label="h2")
    axislegend()
    ratioax = Axis(f[2, 1], aspect = 5.5, xlabel = "GeV", ylabel="data / MC")
    linkxaxes!(ratioax, a)
    hidexdecorations!(a)
    ratiohist!(ratioax, h1, h2; errors=true)
    rowsize!(f.layout, 1, Aspect(1, 0.5))
    f
end
```
![image](https://user-images.githubusercontent.com/5306213/173171792-3c572334-a621-4261-bb31-bc758dc2654f.png)
